### PR TITLE
Reset sprite quadrant placements as part of sprite reset command

### PR DESCRIPTION
**Pull request by janisozaur**
This addresses some sources of desyncs in multiplayer.

@zsilencer I took your commit from https://github.com/janisozaur/OpenRCT2/pull/15/commits/c6fd4b7b1cc8c84d8fce73e755fab6b477504856 and did minor refactor. I have tested it against live server as well running version from before this change, it worked correctly with it, so shouldn't affect non-updated players.

So far I have not triggered any desync while idling but @zsilencer says he managed to get some.

### CODE REVIEW

File: src/world/sprite.c [hunk 1]
```diff
@@ -517,6 +518,32 @@ void sprite_move(sint16 x, sint16 y, sint16 z, rct_sprite* sprite){
...
+		break;
+	case 2:
+		new_x = -new_y + new_x;
+		new_y = (-new_y - start_x) / 2 - z;
+		break;
+	case 3:
+		new_x = new_y + new_x;
+		new_y = (-new_y + start_x) / 2 - z;
+		break;
+	}
```
Comment (IntelOrca) [inline src/world/sprite.c]: The transformation of x, y against rotation is already in sprite_move. Consider moving both instances to a common function.

Comment (janisozaur) [inline src/world/sprite.c]: Actually, if you look closely, this function can be reduced to `sprite->unknown.x = x;` for `x`, `y` and `z`.

**Comment by janisozaur** [comment]
@IntelOrca updated

**Comment by janisozaur** [comment]
Given what this PR changes, it seems that `uncap FPS` option and running in non-headless mode can trigger desyncs. Perhaps the reason I couldn't get any was because I'm mostly testing it headless.

**Comment by IntelOrca** [comment]
Looks fine

**Pull request merged**